ddtrace/contrib/requests/connection.py

```diff
@@ -87,32 +86,36 @@ def _wrap_send(func, instance, args, kwargs):
             propagator = HTTPPropagator()
             propagator.inject(span.context, request.headers)
 
-        # Storing request headers in the span
-        store_request_headers(request.headers, span, config.requests)
-
         response = None
         try:
             response = func(*args, **kwargs)
 
             # Storing response headers in the span. Note that response.headers is not a dict, but an iterable
             # requests custom structure, that we convert to a dict
             if hasattr(response, "headers"):
-                store_response_headers(dict(response.headers), span, config.requests)
+                response_headers = response.headers
+            else:
+                response_headers = None
+            trace_utils.set_http_meta(
+                span, config.requests, request_headers=request.headers, response_headers=response_headers
+            )
             return response
         finally:
             try:
                 status = None
-                if config.requests.trace_query_string:
-                    span.set_tag(http.QUERY_STRING, parsed_uri.query)
                 if response is not None:
                     status = response.status_code
                     # Storing response headers in the span.
                     # Note that response.headers is not a dict, but an iterable
                     # requests custom structure, that we convert to a dict
                     response_headers = dict(getattr(response, "headers", {}))
-                    store_response_headers(response_headers, span, config.requests)
                 trace_utils.set_http_meta(
-                    span, config.requests, method=request.method.upper(), url=sanitized_url, status_code=status
+                    span,
+                    config.requests,
+                    method=request.method.upper(),
+                    url=sanitized_url,
+                    status_code=status,
+                    query=parsed_uri.query,
```

Comment:
Seems like we're missing the response_headers here? A test should probably have caught this 🤔

Reply:
They're stored in lines 95-101 :) using set_http_meta(), we have to store them in a different place in the code

Reply:
ah missed it!